Specs for Book Model

cquiro opened the pull request:
## Summary

- Add gems faker, factory_bot_rails, rspec-rails and database_cleaner to Gemfile and bundle install.
- Install rspec and configure.
- Create config files for factory_bot and database_cleaner under spec/support.
- Write specs to test Book model validations.

## Trello Card

https://trello.com/c/p4XbIViN/22-crear-tests-para-el-modelo-de-book

cquiro committed:
Add gems rspec, factory bot, database cleaner and faker. Run bundle install.

cquiro committed:
Install RSpec. Add configuration files for factory_bot and database_cleaner.

cquiro committed:
Add tests to the book model

- Create a Book factory.

- Write specs to test book validations.

cquiro committed:
Fix rubocop offenses

- Exclude spec_helper and rails helper from all cops.

- Exclude all files under spec from Metrics/BlockLength cop.

cquiro committed:
Create rent model

- Create rent model.
- Add has_many :rents line to book and user models.

cquiro committed:
Create spec files for rent.

- Create rents factory.
- Create users factory.
- Create empty rent_spec.rb file.

cquiro committed:
Fix rubocop offenses

- Add dependent: :destroy to has_many line in book and user.
- Replace double quotes with single quotes in rents and users factories.

lmasello commented on spec/factories/books.rb:
```diff
@@ -0,0 +1,10 @@
+    image { Faker::Placeholdit.image('350x500') }
+    title { Faker::Book.title }
+    publisher { Faker::Book.publisher }
+    year { Faker::Number.between(1450, 2018).to_s }
```
We'll have such old books :D
(This is not a requested change, just a comment)

lmasello commented on spec/models/book_spec.rb:
```diff
@@ -0,0 +1,45 @@
+    expect(book).to be_valid
+  end
+
+  it 'is not valid without a genre' do
```
We can validate the presence of attributes in one line such as the following:
`is_expected.to validate_presence_of(:genre)`.

See: https://relishapp.com/rspec/rspec-core/docs/subject/one-liner-syntax

cquiro committed:
Add specs for rent

- Write specs for rent
- Write validator in rent to make sure end_date is never earlier than
start date.

facucompe commented on spec/models/book_spec.rb:
```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe Book do
```
Besides Lean's comment we also add.
```
subject do
  build(:menu_item)
end

it { is_expected.to be_valid }
```
to check the factory works fine

cquiro committed:
Test book validations with one-liners as per CR

- Add shoulda-matchers gem to Gemfile.

- Require 'shoulda/matchers' in rails_helper.rb.

- Add Shoulda::Matchers configuration to rails_helper.rb.

- Run bundle install.

- Modify validations tests in book_spec to use one-liners.

cquiro committed:
Merge branch create_book_model into test_book_model.

cquiro committed:
Reset db.

cquiro committed:
Merge branch test_book_model into create_rent_model.

cquiro committed:
Use Faker gem in factories

- Use Faker::Date in rents factory.

- Use Faker::Internet.password(8) in users factory.

cquiro committed:
Use one-liners to test validations in rent_spec.rb

cquiro committed:
Validate for presence of rent start and end dates

- Specify 'null: false' in the create rents migration.

- Reset db.

- Remove code to validate presence of user and book in the
rent model.

cquiro committed:
Test custom validation for success

In rent_spec.rb test if custom validation works, not if the error
message returned is as expected.

cquiro committed:
Merge pull request #8 from wolox-training/create_rent_model

Create Rent Model
